### PR TITLE
forked from Genoil cpp-ethereum 117 ,modified CPU thread(can be using with low cpu),and GPU CUDA code

### PR DESCRIPTION
**Pull request by wanyvic**
with my minimg machine, 8-gtx1060  189MH/S

### CODE REVIEW

**Comment by kryoz** [comment]
this won't compile under linux

**Comment by wanyvic** [comment]
@kryoz my code is only optimized for windows.I'm not build it on linux,because it don't need to optimize.you can just replace my libethash-cuda folder with yours on linux.see details https://github.com/Genoil/cpp-ethereum/pull/228

**Comment by ghost** [comment]
@wanyvic Can you check if your improvement is also possible for this repo?
https://github.com/ethereum-mining/ethminer

If you are confused why I am asking about this ... see #268

File: libstratum/EthStratumClient.cpp [hunk 1]
```diff
@@ -520,15 +540,15 @@ bool EthStratumClient::submit(EthashProofOfWork::Solution solution) {
 		string json;
 
 		switch (m_protocol) {
-			case STRATUM_PROTOCOL_STRATUM:
```
Comment (EoD) [inline libstratum/EthStratumClient.cpp]: Unnecessary changes?

**Comment by EoD** [comment]
@wanyvic you have not even one commit message, you might want to add those.

There are also plenty of (unnecessary?) formatting changes in 4d8f908f892a1129a40d7c4ff29fe235910f56ae and b8a7036c76ddfa785c6d3d79c6dbb49c01c86bc4. Are you sure you want to commit those?

I guess this [pack.h](https://github.com/wanyvic/cpp-ethereum/blob/4d8f908f892a1129a40d7c4ff29fe235910f56ae/libethcore/pack.h) is where you "drop" non-Windows support. Is there not an os-agnostic way of doing it?

**Comment by wanyvic** [comment]
@EoD https://github.com/Genoil/cpp-ethereum/pull/250/files look in this way
